Automattic/studio
# Add Sync select site modal

- Closes https://github.com/Automattic/dotcom-forge/issues/9437

## Proposed Changes

- Add Modal to select sites. It users hardcoded sites. We'll need to create another issue to connect it to the API.

<img width="1032" alt="Screenshot 2024-10-16 at 15 40 30" src="https://github.com/user-attachments/assets/e20190af-853a-4625-b6cf-7f4b54eb2265">


## Testing Instructions



- Run `STUDIO_SITE_SYNC=true npm start`
- Click on Sync tab
- Click on Connect site
- Observe a modal opens with the list of your sites
- Try searching and observe the sites filter correctly
- Select a site
- Click connect
- Observe the modal closes. The logic of displaying the list of sites will be implemented on https://github.com/Automattic/dotcom-forge/issues/9393

## Screenshots

**Hover state**

<img width="1032" alt="Screenshot 2024-10-16 at 15 37 39" src="https://github.com/user-attachments/assets/b1ee02ba-2967-4057-9c46-c166d67d8508">

**Selected state**
<img width="1032" alt="Screenshot 2024-10-16 at 15 43 00" src="https://github.com/user-attachments/assets/a70775c8-df8c-4730-aefb-34c3adad1370">


**Not search match**

<img width="1032" alt="Screenshot 2024-10-16 at 15 37 32" src="https://github.com/user-attachments/assets/12d50993-424c-4051-adf5-024e1f0a555d">




## Pre-merge Checklist
